client/environment_test.go

```diff
@@ -415,5 +415,8 @@ func TestMarshalEnvironmentCreateWithoutTemplate(t *testing.T) {
 	var environmentCreateFromJSON EnvironmentCreate
 
 	require.NoError(t, json.Unmarshal(b, &environmentCreateFromJSON))
-	require.Equal(t, environmentCreate, environmentCreateFromJSON)
+
+	environmentCreateWithType := environmentCreate
+	environmentCreateWithType.Type = templateCreate.Type
```

Comment:
needed to update this test. When create an environment with a template inside it, the environment "inherits" the type from the template.

E.g.: if template inside environment is workflow, the environment type is also workflow.